From c7c2d4800a37f314b72df969f0d36a55332bd13d Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 08 Jan 2019 11:35:17 +0200
Subject: [PATCH] [SW-2939] pll: fix delay loop

Since commit:
5e3cdea ([SW-2939] pll: do not initialize the CPU PLL)
we assume the CPU PLL is initialized in BTL and therefore the delay
loop should take it into an account.
Unfortunately, the above commit did not took the delay loop into
account, and it results in lower delays than there should be.
Fix the delay loop to the correct value.

Change-Id: Ie323db3d9033b9c66f1ea6d55f4fcebc6e3de05a
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/pll.c b/pll.c
index 3771ba6..dcaf956 100644
--- a/pll.c
+++ b/pll.c
@@ -29,12 +29,13 @@
 #define CPU_CYCELS_PER_USEC	(1200000000 / USEC_PER_SEC)
 #define BYPASS_CYCLES_PER_USEC	(50000000 / USEC_PER_SEC)
 #define MIN_CYCLES_PER_ITER	5
-static int cpu_pll_bypassed = 1;
 
 #define mmCPU_PLL_BASE		0x7FFC4A2000ull
 #define mmIC_PLL_BASE		0x7FFC4A3000ull
 #define mmPSOC_MME_PLL_BASE	0x7FFCC71000ull
 
+static int cpu_pll_bypassed = 0;
+
 struct hl_pll_cfg {
 	char	*name;
 	void	*base;
